### PR TITLE
[Design2-76] DSCO Alert component

### PR DESCRIPTION
**Pull request by levadadenys**
### [Design2-76] DSCO Alert component

* implemented component
* added tests
* added storybook
* component is now ```Ready for Dev```
* Initial release

https://github.com/code-dot-org/code-dot-org/assets/22244040/71221b7b-a78c-4aa4-a826-f27d2559f864

## Links

- jira ticket: [Design2-76](https://codedotorg.atlassian.net/browse/DESIGN2-76)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/componentLibrary/alert/Alert.story.tsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+GroupOfSizesOfAlerts.args = {
+  components: [
+    {
+      text: 'Alert XS',
+      type: 'primary',
+      size: 'xs',
+      icon: {iconName: 'circle-check'},
+      link: {
+        href: '#',
+        text: 'This is a plink',
```
Comment (kelbyhawn) [inline apps/src/componentLibrary/alert/Alert.story.tsx]: Is `plink` a typo?

File: apps/src/componentLibrary/alert/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# `apps/src/componentLibrary/alert`
+
+## Consuming This Component
+
+This package exports two styled React components: [Alert](Alert.tsx).
```
Comment (kelbyhawn) [inline apps/src/componentLibrary/alert/README.md]: Should this say `one styled React component:`?

**Review by kelbyhawn — APPROVED** [review]
Left a couple copy-related questions but otherwise looks great!

File: apps/src/componentLibrary/alert/Alert.story.tsx (hunk 1 above)
Comment (levadadenys) [reply apps/src/componentLibrary/alert/Alert.story.tsx]: Thanks for spotting! It was intended for testing purposes, but can be deleted now.

File: apps/src/componentLibrary/alert/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# `apps/src/componentLibrary/alert`
+
+## Consuming This Component
+
+This package exports two styled React component: [Alert](Alert.tsx).
```
Comment (kelbyhawn) [inline apps/src/componentLibrary/alert/README.md]: This still says "two", should it be one?
```suggestion
This package exports one styled React component: [Alert](Alert.tsx). 
```

**Review by sanchitmalhotra126 — APPROVED** [review]
LGTM!

**Pull request merged**